File: backend/src/main/java/net/zorphy/backend/site/nobodysperfect/dto/GameRoom.java
package net.zorphy.backend.site.nobodysperfect.dto;

import net.zorphy.backend.site.core.ws.dto.GameRoomBase;
import net.zorphy.backend.site.core.ws.dto.GameRoomMember;

import java.time.Instant;
import java.util.List;

public record GameRoom(
        Instant createdAt,
        String roomId,
        List<GameRoomMember> members,
        GameRoomMember host
) implements GameRoomBase { }

File: frontend/src/app/sites/nobody-is-perfect/dto/GameRoom.ts
import {GameRoomMember} from "../../core/ws/dto/GameRoomMember";
import {GameRoomBase} from "../../core/ws/dto/GameRoomBase";

export interface GameRoom extends GameRoomBase {
    createdAt: string;
    roomId: string;
    members: GameRoomMember[];
    host: GameRoomMember;
}